doc/multi_asic/SONiC_multi_asic_hld.md

```diff
@@ -0,0 +1,937 @@
+BGP docker when comes in asic namespace will remove the default docker route and add it back with lower priority(higher metric value)
+
+For IPv6 sysctl.net.ipv6.conf.all.forwarding was 0 in asic namespace which makes default route learn on local link-address and not on global address. To overcome this and any other possible mismatch in sysctl.net.*
+value between host and asic namespace common file having these values will be used and applied for both host and asic namespace. 
+
+#### 2.4.3.6. BGP configuration
+ When BGP is running at back-end ASICs, some more bgp configurations are required. They are:
+  - Redistribution of connected routes are configured on frontend ASICs
+  - Route reflector is configured on the backend ASICs.
+  - All the internal neighbors are configured *next-hop-self self*
+    
+### 2.4.4. Database Service
+There will be  **N+1** database containers running in a **N**-ASIC device with a single CPU. 
+
+  -  The database container running in the linux host will be called _globalDB_ container in a multi asic platform. The database tables populated here will be system wide attributes like SYSLOG, AAA, TACACS, Mgmt interface and resources like fan, psu, thermal.
+  -  The  ***database{n}*** container will be started in each namespace. _n_  here denotes the ASIC_ID, These databases will be used and updated by applications in that namespace.
+
+Each database container will have it's own **"/var/run/redis{n}/"** directory which contains the database_config.json file, redis unix socket etc. The database_config.json will no longer be a static build time file.
+The database_config.json files are dynamically rendered from the template database_config_json.j2. One database_config.json is generated per database container based on parameters like namespace_id <'', 0, 1, 2, ..n-1>. 
+
+The system wide unique unix socket path is present in the database_config.json. This will be used by connector classes to access the db_tables in different namespaces.  
+    
+A new template file database_global.json.j2 is introduced which is used to generate dynamically a static database_global.json file in the system. This file has references to various database_config.json files used by database containers running on host and in the namespaces.
+
+**Sample database_global.json for 3 ASIC device**
+
+```
+{
+    "INCLUDES" : [
+        {
+            "include" : "../../redis/sonic-db/database_config.json"
+        },
+        {
+            "namespace" : "asic0",
+            "include" : "../../redis0/sonic-db/database_config.json"
+        },
+        {
+            "namespace" : "asic1",
+            "include" : "../../redis1/sonic-db/database_config.json"
+        },
+        {
+            "namespace" : "asic2",
+            "include" : "../../redis2/sonic-db/database_config.json"
+        },
+    ],
+    "VERSION" : "1.0"
+}
+
+```
+**Sample database_config.json for ASIC0**

```

Comment:
Does the design consider multi db support? is this template consider multi db instances ?